polaris/ocean/ice_shelf/ssh_adjustment.py

```diff
@@ -0,0 +1,160 @@
+
+        self.add_input_file(filename='final.nc',
+                            work_dir_target=f'{forward.path}/output.nc')
+        self.add_input_file(filename='init.nc',
+                            work_dir_target=f'{forward.path}/init.nc')
+        self.add_input_file(filename='mesh.nc',
+                            work_dir_target=f'{init.path}/culled_mesh.nc')
+        self.add_output_file(filename='output.nc')
+
+    # no setup() is needed
+
+    def run(self):
+        """
+        Adjust the sea surface height or land-ice pressure to be dynamically
+        consistent with one another.
+        """
+        logger = self.logger
+        config = self.config
+        adjust_variable = config.get('ssh_adjustment', 'adjust_variable')
+        mask_variable = config.get('ssh_adjustment', 'mask_variable')
+        mesh_filename = 'mesh.nc'
+        init_filename = 'init.nc'
+        final_filename = 'final.nc'
+        out_filename = 'output.nc'
+        ds_mesh = xr.open_dataset(mesh_filename)
+        ds_init = xr.open_dataset(init_filename)
+        ds_final = xr.open_dataset(final_filename)
+        ds_out = ds_init.copy()
+
+        if adjust_variable not in ['ssh', 'landIcePressure']:
+            raise ValueError(f"Unknown variable to modify: {adjust_variable}")
+        if mask_variable not in ds_init.keys():
+            raise ValueError(f"Mask variable {mask_variable} is not contained "
+                             f"in {init_filename}")
+
+        logger.info("   * Updating SSH or land-ice pressure")
+
+        # keep the data set with Time for output
+        # and generate these time slices
+        ds_init = ds_init.isel(Time=0)
+        ds_final = ds_final.isel(Time=-1)
+
+        on_a_sphere = ds_out.attrs['on_a_sphere'].lower() == 'yes'
+
+        if 'minLevelCell' in ds_final:
+            minLevelCell = ds_final.minLevelCell - 1
+        else:
+            minLevelCell = ds_mesh.minLevelCell - 1
+
+        init_ssh = ds_init.ssh
+        final_ssh = ds_final.ssh
+        top_density = ds_final.density.isel(nVertLevels=minLevelCell)
+
+        mask = ds_init[mask_variable]
+        delta_ssh = mask * (final_ssh - init_ssh)
+
+        # then, modify the SSH or land-ice pressure
+        if adjust_variable == 'ssh':
+            final_ssh = final_ssh.expand_dims(dim='Time', axis=0)
+            ds_out['ssh'] = final_ssh
+            # also update the landIceDraft variable, which will be used to
+            # compensate for the SSH due to land-ice pressure when
+            # computing sea-surface tilt
+            ds_out['landIceDraft'] = final_ssh
+            # we also need to stretch layerThickness to be compatible with
+            # the new SSH
+            stretch = ((final_ssh + ds_mesh.bottomDepth) /
+                       (init_ssh + ds_mesh.bottomDepth))
+            ds_out['layerThickness'] = ds_out.layerThickness * stretch
+            land_ice_pressure = ds_out.landIcePressure.values
```

Comment:
I know this is ported straight from Compass but it seems like it's only correct for a z-star coordinate.  Maybe that's okay for now but we should keep it in mind.

Reply:
Right, let's just fix this now. I first thought about calling `init_vertical_coord` again with ds_out as an argument, now that this has the new ssh. But I think we actually need to create `polaris.ocean.vertical.update_layer_thickness` that calls each `polaris.ocean.vertical.{vert_cood}.compute_{vert_coord}_layer_thickness` so that only `layerThickness` is altered (and not e.g. `minLevelCell`).

Reply:
@xylar Thoughts about this, when you have a chance. Happy to make a few changes to the vertical module to make this work.

Reply:
Yes, I think we don't want to be changing `minLevelCell` or `maxLevelCell` here so it would be appropriate to have the function you describe.  If you're happy to add it, I think that's great!

Reply:
Wonderful, thanks!